engine/aerodynamics.py

```diff
@@ -12,82 +12,76 @@ def __init__(self, params:vehicle_params.BaseVehicle, logger:engine.Logger):
+
+
         s_dir = -1
         if body_slip < 0:
             s_dir = 1
 
-        # iterates for each aero component
-        for i in range (0, 3):
+        # multiply each sensitivity by the corresponding angle
+        # repeat -> reshape turns angles array into 3x3x3 array with angles in
+        #   the right place for element-wise multiplication
+        angle_sens = np.multiply(angle_sens, np.reshape(np.repeat(angles,9),(3,3,3)))
+
+        # add 1 to turn percent increases into multiplication factor
+        angle_sens += 1
 
-            # sensitivities for Cl, Cd, and Cs
-            Cl_sens = (1 + self.bs_sens[i][0] * abs(body_slip * self.rad_to_deg)) * (1 + self.p_sens[i][0][p_dir] *
-                        abs(pitch * self.rad_to_deg)) * (1 + self.r_sens[i][0] * abs(roll * self.rad_to_deg))
-            Cd_sens = (1 + self.bs_sens[i][1] * abs(body_slip * self.rad_to_deg)) * (1 + self.p_sens[i][1][p_dir] *
-                        abs(pitch * self.rad_to_deg)) * (1 + self.r_sens[i][1] * abs(roll * self.rad_to_deg))
-            Cs_sens = (1 + self.p_sens[i][2][p_dir] * abs(pitch * self.rad_to_deg)) * (1 + self.r_sens[i][2] *
-                        abs(roll * self.rad_to_deg))
+        # multiply all sensitivities for given part and force together
+        angle_sens = np.prod(angle_sens, axis = 0)
 
-            # Cl, Cd, and Cs for the compenent
-            ClA_part = self.ClA[i] * Cl_sens
-            CdA_part = self.CdA[i] * Cd_sens
-            CsA_part = self.CsA[i] * Cs_sens
+        # multiply lift, drag, and sideforce coefficients by sensitivities
+        coefs = np.multiply(angle_sens, coefs.T)
 
-            # calculate force in each direction
-            Fl_part = 0.5 * self.__air_density * ClA_part * x_dot ** 2
-            Fd_part = 0.5 * self.__air_density * CdA_part * x_dot ** 2
-            Fs_part = 0.5 * self.__air_density * CsA_part * (x_dot * np.tan(body_slip)) ** 2 * s_dir
+        # calculate force arrays for each direction
+        Fl_part = 0.5 * self.__air_density * coefs[:,0] * x_dot ** 2
+        Fd_part = 0.5 * self.__air_density * coefs[:,1] * x_dot ** 2
+        Fs_part = 0.5 * self.__air_density * coefs[:,2] * (x_dot * np.tan(body_slip/rad_to_deg)) ** 2 * s_dir
+
+        #
+        part_force = np.array([-Fd_part, Fs_part, -Fl_part])
+
+        forces = np.array([-np.sum(Fd_part), np.sum(Fs_part), -np.sum(Fl_part)])
+        moments = np.cross(self.vehicle_params.CoP[0], part_force.T[0]) \
+                + np.cross(self.vehicle_params.CoP[1], part_force.T[1]) \
+                + np.cross(self.vehicle_params.CoP[2], part_force.T[2])
 
-            part_force = np.array([-Fd_part, Fs_part, -Fl_part])
-            forces = np.add(forces, part_force)
-            moments = np.add(moments, np.cross(self.CoP[i], part_force))
 
         # account for drag from rest of car
         drag_no_aero = 0.5 * self.__air_density * self.vehicle_params.CdA0 * x_dot ** 2
         forces[0] -= drag_no_aero
```

Comment:
@sarahfedorchak Can we make this and the other similar equation just one line?

Reply:
I don't think that would be helpful because the equations aren't exactly the same.

Reply:
I mean the forces[0] and forces[1]